### PR TITLE
Version binding for numpy

### PR DESCRIPTION
**Pull request by dkfellows**
From https://numpy.org/devdocs/release/1.20.0-notes.html

> support for Python 3.6 has been dropped

The real details are in https://numpy.org/neps/nep-0029-deprecation_policy.html

### CODE REVIEW

**Comment by rowleya** [comment]
Minor point:  if you want to allow the version to evolve for later Python versions, you can do this:
https://stackoverflow.com/questions/19559247/requirements-txt-depending-on-python-version

I don't actually know if we do want to allow this though...

**Comment by dkfellows** [comment]
I was already working on that. I knew I'd already done it somewhere… but I had to look up where. 😉

File: doc/source/conf.py [hunk 1]
```diff
@@ -56,7 +56,7 @@
 
 intersphinx_mapping = {
     'python': ('https://docs.python.org/3.6', None),
-    'numpy': ("https://numpy.org/doc/stable/", None),
+    'numpy': ("https://numpy.org/doc/1.19/", None),
```
Comment (Christian-B) [inline doc/source/conf.py:59]: I would rather have seen us moving sphinx to a later python version but that is out fo scope of this PR

**Review by Christian-B — APPROVED** [review]
Merge when green!

File: doc/source/conf.py (hunk 1 above)
Comment (dkfellows) [reply doc/source/conf.py:59]: That controls what version of the numpy documentation we point to in our types.

**Comment by Christian-B** [comment]
If this works, As a second PR see is adding a version for 2.7 gets setup to work without the need to install requirement. But 1, AFTER this PR and 2 only if easy

**Comment by dkfellows** [comment]
Feel free to write the PR. The version range binding trickery I've used here is accepted in both `requirements.txt` and `setup.py`; it's totally Python standard and has been for ages.

**Comment by Christian-B** [comment]
> Feel free to write the PR. The version range binding trickery I've used here is accepted in both `requirements.txt` and `setup.py`; it's totally Python standard and has been for ages.

A quick check:
https://github.com/SpiNNakerManchester/PACMAN/runs/1509289122?check_suite_focus=true

Shows there is a lot more than just numpy broken in python 2,7 so not worth it.

**Pull request merged**